app/src/main/java/ch/epfl/sweng/erpa/activities/GameViewerActivity.java

```diff
@@ -0,0 +1,95 @@
+package ch.epfl.sweng.erpa.activities;
+
+
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.TextView;
+
+import com.annimon.stream.Optional;
+
+import java.util.NoSuchElementException;
+
+import javax.inject.Inject;
+
+import butterknife.BindView;
+import butterknife.ButterKnife;
+import ch.epfl.sweng.erpa.R;
+import ch.epfl.sweng.erpa.model.Game;
+import ch.epfl.sweng.erpa.services.RemoteServicesProvider;
+
+import static android.content.ContentValues.TAG;
+
+public class GameViewerActivity extends DependencyConfigurationAgnosticActivity {
+
+    public static final String EXTRA_GAME_KEY = "game";
+    @Inject
+    RemoteServicesProvider rsp;
+
+    @BindView(R.id.titleTextView)
+    TextView title;
+    @BindView(R.id.descriptionTextView)
+    TextView description;
+    @BindView(R.id.gmTextView)
+    TextView gmName;
+    @BindView(R.id.universeTextView)
+    TextView universe;
+    @BindView(R.id.generalInfoTextView)
+    TextView generalInfo;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_game_viewer);
+        ButterKnife.bind(this);
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        Optional<Game> optGame;
+        String gameId = getGameId();
+        optGame = rsp.getGameService().getGame(gameId);
+
+        if (optGame.isPresent()) {
+            Game game = optGame.get();
+            updateFields(game);
+        } else {
+            Log.d(TAG, "onResume: could not find game in database. Exiting", new NoSuchElementException());
+            finish();
+        }
+    }
+
+
+    private String getGameId() {
+
+        String gameId = getIntent().getStringExtra(GameViewerActivity.EXTRA_GAME_KEY);
+        if (gameId == null) {
+            Exception thrown = new IllegalArgumentException("Game Id not found");
+            Log.d(TAG, "GameViewerActivity: no game id passed with intent", thrown);
+            finish();
+            throw new IllegalArgumentException();
+        } else {
+            return gameId;
+        }
+    }
+
+    private void updateFields(Game game) {
+        title.setText(game.getName());
+        description.setText(game.getDescription());
+        gmName.setText(game.getGmUid());
+        universe.setText(game.getUniverse());
+        Log.d(TAG, "onResume: Successfully fetched game");
+
+        String miscInfo = String.format("Difficulty: %s\n" +
+                        "%s\n" +
+                        "Number of sessions: %s\n" +
+                        "Session length: %s",
+                game.getDifficulty(), game.getType(), game.getNumSessions(), game.getSessionLength());
+
+        generalInfo.setText(miscInfo);
```

Comment:
Why a single field containing all these values. I think it'd be better if we had an element for each field